service/src/main/java/com/derpgroup/derpwizard/resource/AlexaResource.java

```diff
@@ -157,62 +159,28 @@ public SpeechletResponseEnvelope doAlexaRequest(@NotNull @Valid SpeechletRequest
         else{
           card = null;
         }
-        
-        outputSpeech = new SsmlOutputSpeech();
-        outputSpeech.setSsml(serviceOutput.getVoiceOutput().getSsmltext());
-        speechletResponse.setShouldEndSession(serviceOutput.isConversationEnded());
-        break;
+        if(serviceOutput.getDelayedVoiceOutput() !=null && StringUtils.isNotEmpty(serviceOutput.getDelayedVoiceOutput().getSsmltext())){
+            reprompt = new Reprompt();
+            SsmlOutputSpeech repromptSpeech = new SsmlOutputSpeech();
+            repromptSpeech.setSsml("<speak>"+serviceOutput.getDelayedVoiceOutput().getSsmltext()+"</speak>");
+            reprompt.setOutputSpeech(repromptSpeech);
+          }
+
+          outputSpeech = new SsmlOutputSpeech();
+          outputSpeech.setSsml("<speak>"+serviceOutput.getVoiceOutput().getSsmltext()+"</speak>");
+          shouldEndSession = serviceOutput.isConversationEnded();
+          break;
       }
+
+      Map<String,Object> sessionAttributes = mapper.convertValue(outputMetadata, new TypeReference<Map<String,Object>>(){});
       
-      speechletResponse.setOutputSpeech(outputSpeech);
-      speechletResponse.setCard(card);
-      responseEnvelope.setResponse(speechletResponse);
-      responseEnvelope.setVersion(ALEXA_VERSION);
-  
-      return responseEnvelope;
+      return AlexaUtils.buildOutput(outputSpeech, card, reprompt, shouldEndSession, sessionAttributes);
     }catch(DerpwizardException e){
       LOG.debug(e.getMessage());
-      return new DerpwizardExceptionAlexaWrapper(e, "1.0",mapper.convertValue(outputMetadata, new TypeReference<Map<String,Object>>(){}));
+      return new DerpwizardExceptionAlexaWrapper(e, ALEXA_VERSION,mapper.convertValue(outputMetadata, new TypeReference<Map<String,Object>>(){}));
```

Comment:
I dunno, probably our skill's version, but we don't track that anywhere internally.